src/radiation_system.hpp

```diff
@@ -687,6 +693,104 @@ AMREX_GPU_DEVICE auto RadSystem<problem_t>::ComputeCellOpticalDepth(const quokka
 						       // return 0.5*(tau_L + tau_R); // arithmetic mean
 }
 
+template <typename problem_t>
+template <FluxDir DIR, typename TARRAY>
+AMREX_GPU_DEVICE void RadSystem<problem_t>::ComputeRadPressure(TARRAY &F, double &S, const double erad, const double Fx, const double Fy, const double Fz,
```

Comment:
I really think this function should return `std::tuple<TARRAY, double>` instead of `void`.